openstack/identity/v3/roles/requests.go

```diff
@@ -148,3 +148,74 @@ func ListAssignments(client *gophercloud.ServiceClient, opts ListAssignmentsOpts
 		return RoleAssignmentPage{pagination.LinkedPageBase{PageResult: r}}
 	})
 }
+
+// AssignOpts provides options to assign a role
+type AssignOpts struct {
```

Comment:
I agree that `AssignOpts` may be counter-intuitive to the users calling `Unassign`.
Since it is not a lot of duplicate code and is a better user-facing design, I will make change to the option 1 for now. 
If there is any objection to this, please let me know.

Reply:
> Since it is not a lot of duplicate code and is a better user-facing design, I will make change to the option 1 for now.

This sounds good to me. I've been revisiting this throughout the day and Option 1 seems like the most straightforward solution.